sdk/src/impl/position-impl.ts

```diff
@@ -189,6 +190,75 @@ export class PositionImpl implements Position {
     return txBuilder;
   }
 
+  async collectFees(
+    resolveATA: boolean = true,
+    destinationWallet?: Address,
+    positionWallet?: Address,
+    ataPayer?: Address
+  ): Promise<TransactionBuilder> {
+    const [destinationWalletKey, positionWalletKey, ataPayerKey] = AddressUtil.toPubKeys([
+      destinationWallet ?? this.ctx.wallet.publicKey,
+      positionWallet ?? this.ctx.wallet.publicKey,
+      ataPayer ?? this.ctx.wallet.publicKey,
+    ]);
+
+    const whirlpool = await this.fetcher.getPool(this.data.whirlpool, true);
+    if (!whirlpool) {
+      throw new Error("Unable to fetch whirlpool for this position.");
+    }
+
+    // Question @meep: Can either fee_owed_a or fee_owed_b be 0 while the other is non-zero?
+    //                 If so, will the collect_fees IX fail since it just blindly calls a token transfer CPI?
+
+    // TODO(atamari): Resolve the following questions:
+    // 1. Question mentioned above
+    // 2. Question asked on discord re: stale on-chain values of fee_owed_a and fee_owed_b preventing valid fee collection
```

Comment:
for the most updated value, client would have to call `updateFeesAndRewardsIx` to update reward/fee values on-chain.

If not, the collect call would only fetch the outstanding amount in the account. Any 'technically owed but not updated' amount will be reflected the next time the user adjusts liquidity or call updateFee/rewards.